SafeguardSessions/Scripts/RunUnitTests.ps1

```diff
@@ -10,17 +7,79 @@ $dir = Split-Path $scriptpath
 
 . $dir\PreCheck.ps1
 
-if ($null -eq $Output) {
-    $Output = "short"
+function LogError {
+    param (
+        $testResult
+    )
+    Write-Host "$($testResult.Name): " -NoNewline
+    Write-Host "Error" -ForegroundColor red
+    Write-Host "Message: " -NoNewline -ForegroundColor red
+    Write-Host $testResult.Error.Message
+    Write-Host "Details: " -NoNewline -ForegroundColor red
+    Write-Host $testResult.Error.Details
+
+}
+
+function LogSuccess {
+    param (
+        $testResult
+    )
+    Write-Host "$($testResult.Name): " -NoNewline
+    Write-Host  "Success" -ForegroundColor green
+}
+
+function LogFail {
+    param (
+        $testResult
+    )
+    Write-Host "$($testResult.Name): " -NoNewline
+    Write-Host "Fail" -ForegroundColor red
+    for (($i = 1); $i -lt $testResult.Output.Details.Length; $i++){
+        $cleanedTestResult = $testResult.Output[$i].Details.Trim()
+        # remove opening and closing parenthesis and split string to expected and actual output
+        $resultArray = $cleanedTestResult.Substring(1, $cleanedTestResult.Length-2) -split "<>" 
+        $expectedResult = $resultArray[0].Trim()
+        $actualResult = $resultArray[1].Trim()
+        Write-Host  "Assert Failed: " -NoNewline -ForegroundColor red
+        Write-Host $testResult.Output[$i].Notes 
+        Write-Host  "Expected Output: " -NoNewline -ForegroundColor green
+        Write-Host $expectedResult
+        Write-Host  "Actual Output: " -NoNewline -ForegroundColor red
+        Write-Host $actualResult
+    }
+    
 }
 
-if ($null -eq $TestFile) {
-    $TestFile = $config.unitTestsPath
+$Output = "short"
+if ($null -eq $TestFile){
+    $testInput = $config.unitTestsPath
+} else {
+    $testInput = $TestFile
 }
 

```

Comment:
Can log here something like:
`Running unit tests in $testInput`

When we run `make unit-tests` it may take a while until we get feedback, so this reinforcing log would be a good feedback for the developer that the process is started.

Reply:
ACK